mtbl/reader.c

```diff
@@ -422,6 +422,32 @@ reader_iter_free(void *v)
 	}
 }
 
+static bool
+needs_index_seek(struct reader_iter *it, const uint8_t *seek_key, size_t len_seek_key)
+{
+	const uint8_t *key, *val;
+	size_t len_key, len_val;
+
+	if (it->first || it->b == NULL)
+		return true;
+
+	if (!block_iter_get(it->bi, &key, &len_key, &val, &len_val))
+		return true;
+
+	/* backward seek */
+	if (bytes_compare(key, len_key, seek_key, len_seek_key) > 0)
+		return true;
+
+	if (!block_iter_get(it->index_iter, &key, &len_key, &val, &len_val))
+		return true;
+
+	/* seek after end of block */
+	if (bytes_compare(key, len_key, seek_key, len_seek_key) < 0)
```

Comment:
Minor nitpick, but I think removing the newline between the related blocks and making the comments more verbose would help future travelers into this codebase.

Reply:
Fixed, rolled into new version of commit.